### ISSUE TITLE
Metricbeat windows service startup time

### ISSUE THREAD

**Issue by trevorndodds** [elastic/beats#2817]
Metricbeat startup time as a service varies on different windows systems. I'm running into times when the service is unable to start within the standard windows timeout. Is it possible to try start metricbeat faster, similar to how filebeat starts? 

This has been the same for all windows metricbeat releases.

**Comment by tsg** [comment]
That's weird, it seems to start very quickly for me.

Can you try starting it in foreground, and see if it's fast that way:

```
.\metricbeat.exe -e
```

**Comment by trevorndodds** [comment]
It seems that on start the helper.go is checking over each process this seems to take a while. I've pasted the time below. This particular server is idle, but the same can be reproduced if you have a really busy server, with all CPUs busy, metricbeat will never start until the server is mostly idle.

2016/10/21 13:10:24.055015 beat.go:264: INFO Home path:
2016/10/21 13:10:24.071015 metricbeat.go:25: INFO Register
2016/10/21 13:10:24.441015 helper.go:271: DBG  Skip process pid=0: error getting process state for pid=0: OpenProcess fails with The parameter is incorrect.
2016/10/21 13:10:24.442015 helper.go:271: DBG  Skip process pid=428: error getting process state for pid=428: OpenProcess fails with Access is denied.
...
...
2016/10/21 13:10:47.917015 helper.go:271: DBG  Skip process pid=155292: error getting process state for pid=155292: OpenProcess fails with Access is denied.
2016/10/21 13:10:48.560015 beat.go:204: INFO metricbeat start running.

The default windows service startup timeout is about 30 seconds, from above it's close. I have debug enabled in my logs, but when running as a service I only have the below two lines and that's as far as it gets before the service dies.
2016-10-21T09:20:35-04:00 DBG  Skip process pid=0: error getting process state for pid=0: OpenProcess fails with The parameter is incorrect.
2016-10-21T09:20:35-04:00 DBG  Skip process pid=4: error getting process state for pid=4: OpenProcess fails with Access is denied.

**Comment by oazabir** [comment]
Metricbeat as a service does not start at all unless I disable the process stat collection . Running Windows 2008 R2 Standard Edition. 
When I run it from command prompt manually, I get this error message:

```
2016/12/08 10:35:40.090705 helper.go:276: ERR Error getting process details pid=
1632: error getting process mem for pid=1632: OpenProcess fails with Access is d
enied.
```

But it does work and send events for processes when run from command prompt. 

But it does not work when I try to start it as a service. The service user is Local System.

**Comment by andrewkroh** [comment]
This should be fixed by #3056 which you can test by running the snapshot build from [here](https://beats-nightlies.s3.amazonaws.com/index.html?prefix=metricbeat/). The fix hasn't been released yet.

**Comment by andrewkroh** [comment]
Duplicate of #3018

**Comment by trevorndodds** [comment]
Thanks, I'll test the snapshot build.
